scanner/client/options.go

```diff
@@ -0,0 +1,63 @@
+package client
+
+import (
+	"github.com/stackrox/rox/pkg/mtls"
+)
+
+var defaultOptions = options{
+	mtlsSubject:   mtls.ScannerSubject,
+	address:       ":8443",
+	serverName:    "scanner-v4.stackrox",
+	withTLSVerify: true,
+}
+
+// Option configures the options to create a scanner client.
+type Option func(*options)
+
+type options struct {
+	mtlsSubject   mtls.Subject
+	address       string
+	serverName    string
+	withTLSVerify bool
+}
+
+// GetCA loads and instantiates a Stackrox CA from the options specified.
+func (o *options) GetCA() (mtls.CA, error) {
+	// Options only support the default CA for now.
+	return mtls.LoadDefaultCA()
+}
+
+// WithSubject specifies the mTLS subject to use.
+func WithSubject(subject mtls.Subject) Option {
+	return func(o *options) {
+		o.mtlsSubject = subject
+	}
+}
+
+// WithServerName specifies the mTLS server name used to verify the server's certificate.
+func WithServerName(serverName string) Option {
+	return func(o *options) {
+		o.serverName = serverName
+	}
+}
+
+// WithoutTLSVerify disables TLS verification, and don't read or use client
+// certificates (mTLS).
+func WithoutTLSVerify(o *options) {
```

Comment:
I understand your point. This consistency makes the usage of the options uniform by artificially requiring a function in some cases when it's unnecessary. But, I prefer to need a function in an option only when required because it makes it explicit. Callable options require information from the caller, which is passed as arguments when constructing the option. Non-callables are like flags. Think of `$ foobar-command --without-tls --output /tmp/foobar`.